scripts/keeper.ts

```diff
@@ -161,11 +132,45 @@ async function flushesToBeFinalized(
     });
   }
 
-  const logMessageEvents = await getSettleMessageEvents();
+  const logMessageEvents = await getLogMessageEvents();
   return Promise.all(
-    logMessageEvents.filter(async (event: Event) => {
+    logMessageEvents.filter(async (event: LogMessageToL1Event) => {
       const consumedMessageEvents = await getConsumedSettleMessageEvents(event);
       return consumedMessageEvents.length === 0;
     })
   );
 }
+
+export async function finalizeFlush(config: Config) {
+  const l1Signer = getL1Signer(config);
+  const l2Signer = getL2Signer(config);
+
+  const starknet = await getL1ContractAt<Starknet>(
+    l1Signer,
+    "Starknet",
+    config.starknetAddress
+  );
+  const l2TeleportGateway = await getL2ContractAt<l2_dai_teleport_gateway>(
+    l2Signer,
+    "l2_dai_teleport_gateway",
+    config.l2TeleportGatewayAddress
+  );
+  const l1TeleportGatewayAddress = await getL1TeleportGatewayAddress(l2TeleportGateway);
+  const l1TeleportGateway = await getL1ContractAt<L1DAITeleportGateway>(
+    l1Signer,
+    "L1DAITeleportGateway",
+    l1TeleportGatewayAddress
+  );
+
+  const flushes = await flushesToBeFinalized(l1TeleportGatewayAddress, starknet, l1Signer, config);
+  for (let i = 1; i < flushes.length; i++) {
+    const flush = flushes[i];
+    console.log("Sending `finalizeFlush` transaction");
+    const tx = await l1TeleportGateway.finalizeFlush(
+      cairoShortStringToBytes32(flush.args.payload[1]),
+      flush.args.payload[2]
```

Comment:
Add gasPrice management.

Reply:
Usually one wants to be able to mange gas price on l1. For one time scripts overriding gasPrice is usually enough. For an autonomous keeper strategy should be more elaborate. I am not sure how to do it in context of EIP-1559. Let's turn this problem into a separate task.